src/panels/MissionsPanel/MissionsPanel.tsx

```diff
@@ -1,13 +1,41 @@
-import { Stack, Text, ThemeIcon, Title } from '@mantine/core';
+import { Box, Select, Stack, Text, ThemeIcon, Title } from '@mantine/core';
 
+import { useOpenSpaceApi } from '@/api/hooks';
 import { RocketLaunchIcon } from '@/icons/icons';
-import { useAppSelector } from '@/redux/hooks';
+import { useAppDispatch, useAppSelector } from '@/redux/hooks';
+import { setSelectedMission } from '@/redux/missions/missionsSlice';
 
 import { MissionContent } from './MissionContent';
 
 export function MissionsPanel() {
-  const hasMission = useAppSelector((state) => state.missions.isInitialized);
-  const mission = useAppSelector((state) => state.missions.data.missions[0]);
+  const {
+    isInitialized: hasMission,
+    data,
+    selectedMissionIdentifier
+  } = useAppSelector((state) => state.missions);
+
+  const luaApi = useOpenSpaceApi();
+  const dispatch = useAppDispatch();
+
+  let [mission] = data.missions;
+  // If a new missions is added, the missions order might have changed so we try to find
+  // the last one viewed
+  if (hasMission && selectedMissionIdentifier) {
+    const selectedMission = data.missions.find(
+      (_mission) => _mission.identifier === selectedMissionIdentifier
+    );
+    if (selectedMission) {
+      mission = selectedMission;
+    }
+  }
```

Comment:
It seems like the selectedMission takes precedence over getting the first mission, so how about we make this a function / hook and return the index 0 in an else statement? That would make it clearer that we first check for the selected mission and then as a fallback the first element in the array. Plus the hook could return { hasMission, mission } and it would look kinda neat I think
```suggestion
  if (hasMission && selectedMissionIdentifier) {
    const selectedMission = data.missions.find(
      (_mission) => _mission.identifier === selectedMissionIdentifier
    );
    if (selectedMission) {
      mission = selectedMission;
    }
    else {
        mission = data.missions.at(0);
    }
  }
```

Reply:
Checkout the new hook pls and thx